### PR TITLE
Recognize HTTP/2 traffic

### PR DESCRIPTION
**Pull request by mgritter**
Added a (failing) unit test for HTTP/1 parsing.
New ParsedNetworkTraffic types for QUIC and HTTP/2 handshakes.
Parser factory that recognizes HTTP/2 connection preface.  (This comes at the start of the binary format, after upgrade from HTTP/1.1 is successful or the client is connecting to a known HTTP/2-capable server.)
Parser that simply drops everything associated with a HTTP/2 connection.
Unit tests for HTTP/2.
Counters for HTTP/2 and QUIC.

### CODE REVIEW

File: client_telemetry/pcap_stats.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ type PacketCounts struct {
 	HTTPRequests  int `json:"http_requests"`
 	HTTPResponses int `json:"http_responses"`
 	TLSHello      int `json:"tls_hello"`
+	HTTP2Preface  int `json:"http2_preface"`
+	QUICHandshake int `json:"quic_handshake"`
```
Comment (liujed) [inline client_telemetry/pcap_stats.go:17]: `TLSHello` is the odd duck, but the other counts have pluralized names, which also feels more natural.
```suggestion
	HTTP2Prefaces  int `json:"http2_prefaces"`
	QUICHandshakes int `json:"quic_handshakes"`
```

Comment (liujed) [inline client_telemetry/pcap_stats.go:17]: Is this used?

File: akinet/net_traffic.go [hunk 2]
```diff
@@ -374,6 +374,23 @@ func (tls *TLSHandshakeMetadata) ApplicationLatencyMeasurable() bool {
...
+func (HTTP2ConnectionPreface) ReleaseBuffers()           {}
+
+// Represents an observed QUIC handshake (initial packet).
+// Currently empty because we're only interested in the presence
+// of QUIC traffic, not its payload.
+type QUICHandshakeMetadata struct {
+}
+
+func (QUICHandshakeMetadata) implParsedNetworkContent() {}
+func (QUICHandshakeMetadata) ReleaseBuffers()           {}
```
Comment (liujed) [inline akinet/net_traffic.go:392]: Is this used?

**Review by liujed — APPROVED** [review]

File: akinet/net_traffic.go (hunk 2 above)
Comment (mgritter) [reply akinet/net_traffic.go:392]: Not yet, it was originally my next task but it's been deprioritized since this part took so long.

**Pull request merged**